feat(ColorPicker): add ColorPicker component

MEsteves22 opened the pull request:
- A bunch of new properties were added to meet the DS5 specifications. They are listed in the migration guide.
- New stories were added to showcase all the different use cases and variants.
- The last story shows how the component can be customised to meet the DS3 specifications. 
- **As discussed, the dropdown and alpha features were not implemented for now since we need more specifications. The drag feature to remove the saved colors was also not implemented. A button was added instead to remove the color.**


<img width="282" alt="Screenshot 2023-05-12 at 12 55 28" src="https://github.com/lumada-design/hv-uikit-react/assets/43220251/b87f4690-2a45-4c1b-ab1b-e89d6863848b">

HQFOX commented:
if you try to save a color without touching the input it saves a null or undefined value i think, it makes this weird behavior , maybe a default value should be added
![image](https://github.com/lumada-design/hv-uikit-react/assets/19229133/64a3d51f-544e-4a04-9e9b-267dbfe256ab)

MEsteves22 commented on packages/core/src/components/BaseInput/BaseInput.styles.tsx:
```diff
@@ -1,29 +1,28 @@
-  }) => ({
+import { CSSInterpolation } from "@emotion/serialize";
+
+export const styles: {
+  root: CSSInterpolation;
+  inputBorderContainer: CSSInterpolation;
+  disabled: CSSInterpolation;
+  invalid: CSSInterpolation;
+  resizable: CSSInterpolation;
+  readOnly: CSSInterpolation;
+  inputRoot: CSSInterpolation;
+  inputRootInvalid: CSSInterpolation;
+  inputRootReadOnly: CSSInterpolation;
+  inputRootFocused: CSSInterpolation;
+  inputRootDisabled: CSSInterpolation;
+  inputRootMultiline: CSSInterpolation;
+  input: CSSInterpolation;
+  inputResizable: CSSInterpolation;
+  inputDisabled: CSSInterpolation;
+  inputReadOnly: CSSInterpolation;
+} = {
```
Yes, it's the same keys so I'm going to change this to use `HvBaseInputClasses` instead 🙏

MEsteves22 commented:
I updated my PR with the proposed changes. Regarding the TypeScript improvements for the classes, I'll open a new PR to implement the changes to the remaining components  🙏  I only updated the components I had already modified on this PR prior to the review.
